File: airesources/Java/MyBot.java
import hlt.*;

import java.util.ArrayList;

public class MyBot {

    public static void main(final String[] args) {
        final Networking networking = new Networking();
        final GameMap gameMap = networking.initialize("Tamagocchi");

        // We now have 1 full minute to analyse the initial map.
        final String initialMapIntelligence =
                "width: " + gameMap.getWidth() +
                "; height: " + gameMap.getHeight() +
                "; players: " + gameMap.getAllPlayers().size() +
                "; planets: " + gameMap.getAllPlanets().size();
        Log.log(initialMapIntelligence);

        final ArrayList<Move> moveList = new ArrayList<>();
        for (;;) {
            moveList.clear();
            networking.updateMap(gameMap);

            for (final Ship ship : gameMap.getMyPlayer().getShips().values()) {
                if (ship.getDockingStatus() != Ship.DockingStatus.Undocked) {
                    continue;
                }

                for (final Planet planet : gameMap.getAllPlanets().values()) {
                    if (planet.isOwned()) {
                        continue;
                    }

                    if (ship.canDock(planet)) {
                        moveList.add(new DockMove(ship, planet));
                        break;
                    }

                    final ThrustMove newThrustMove = Navigation.navigateShipToDock(gameMap, ship, planet, Constants.MAX_SPEED/2);
                    if (newThrustMove != null) {
                        moveList.add(newThrustMove);
                    }

                    break;
                }
            }
            Networking.sendMoves(moveList);
        }
    }
}

File: airesources/dart/MyBot.dart
import 'lib/hlt.dart';


void main() {
  final net = new Networking();
  final map = net.initialize("Jane");

  // From here until the next map update,
  // 1 minute is allowed to analyse the map.
  log("width:${map.width}, height:${map.height}, players:${map.players.length}, planets:${map.planets.length}");

  final moves = new List<Move>();
  while (true) {
    moves.clear();
    net.updateMap(map);

    for (final ship in map.myShips) {
      if (ship.dockingStatus != DockingStatus.UNDOCKED) {
        continue;
      }

      for (final planet in map.planets) {
        if (planet.isOwned()) {
          continue;
        }

        if (ship.canDock(planet)) {
          moves.add(new DockMove(ship.id));
          break;
        }

        final ThrustMove newThrustMove = navigateShipToDock(map, ship, planet, MAX_SPEED~/2);
        if (newThrustMove != null) {
            moves.add(newThrustMove);
        }

        break;
      }
    }

    net.writeMoves(moves);
  }
}
